### DISCUSSION TITLE
6.7.0 (2023-06-01)

### DISCUSSION THREAD

**Discussion by manusa** [fabric8io/kubernetes-client#5197] [Announcements]
#### Bugs
- Fix #5117: corrected the trace httpclient logging of large response bodies
- Fix #5125: TLS 1.3 only should be supported
- Fix #5126: fallback to changeit only if null/empty does not work
- Fix #5145: [java-generator] handle `additionalProperties: true` emitting a field of type `AnyType`
- Fix #5152: preventing JDK WebSocket errors from terminating watches and improving watch termination and its logging
- Fix #5164: [java-generator] handle more special characters in field names

#### Improvements
- Fix #1335: HttpClient Factory additionalConfig consistently applied for all client types

#### Dependency Upgrade
- Fix #4989: Upgrade Tekton Model to v0.47.0
- Fix #5107: The Camel-k extension has been deprecated in favor of the official release of the generated one

#### New Features
- Fix #4184: Add utility methods for creating ConfigMap from files/directories in KubernetesResourceUtil
- Fix #4829: Gradle Plugin for Java Generation from CRD
- Fix #5086: Allow for supporting socks proxies via proxy urls with the socks4 or socks5 protocol.  Note that the JDK client does not support socks, and the Jetty client does not support socks5

#### _**Note**_: Breaking changes
- Fix #4911: Config/RequestConfig.scaleTimeout has been deprecated along with Scalable.scale(count, wait) and DeployableScalableResource.deployLatest(wait). withTimeout may be called before the operation to control the timeout.
- Fix #4911: Config/RequestConfig.websocketTimeout has been removed. Config/RequestConfig.requestTimeout will be used for websocket connection timeouts.
- Fix #4911: HttpClient api/building changes - writeTimeout has been removed, readTimeout has moved to the HttpRequest
- Fix #4662: removed deprecated classes/methods:  ReflectUtils, ReplaceValueStream, ParameterNamespaceListVisitFromServerGetDeleteRecreateWaitApplicable, ResourceCompare, and Serialization methods taking parameters
- Fix #4662: deprecated serialization static logic:  several IOHelpers methods, Serialization methods, such as access to the static jsonMapper.  Please use KubernetesSerialization methods instead.
- Fix #4662: deprecated Helper.getAnnotationValue, use HasMetadata methods instead.
- Fix #5125: support for TLSv1.3 is now enabled by default
- Fix #5125: usage of TlsVersion.TLS_1_1, TLS_1_0, and SSL_3_0 have been deprecated
- Fix #1335: The JDK and OkHttp clients will default to using the VM's standard configuration for proxies if an applicable proxy configuration is not found in the Kubernetes client Config
 
#### New Contributors
* @dziemba made their first contribution in https://github.com/fabric8io/kubernetes-client/pull/4989
* @rissen made their first contribution in https://github.com/fabric8io/kubernetes-client/pull/5045
* @Indrranil made their first contribution in https://github.com/fabric8io/kubernetes-client/pull/5157
* @miriSch made their first contribution in https://github.com/fabric8io/kubernetes-client/pull/5165

**Full Changelog**: https://github.com/fabric8io/kubernetes-client/compare/v6.6.2...v6.7.0

<hr /><em>This discussion was created from the release <a href='https://github.com/fabric8io/kubernetes-client/releases/tag/v6.7.0'>6.7.0 (2023-06-01)</a>.</em>

**Comment by testagainforperfection** [comment]
Hi Team,

With the latest updates coming in quite frequently, can we please keep updating the documentation for user reference as well?

For example with the deprecation of createOrReplace(), it is suggested to use ServerSideApply(). But due to the lack of documentation, I am unable to get it to work.

**Reply by manusa** [reply]
Reflected in #5176 

Additional comments on #5215

**Reply by manusa** [reply]
> But due to the lack of documentation, I am unable to get it to work.

Could you please elaborate your problem?

**Reply by rohanKanojia** [reply]
> For example with the deprecation of createOrReplace(), it is suggested to use ServerSideApply(). But due to the lack of documentation, I am unable to get it to work.

Which documentation are you referring to? Is it javadocs or doc/CHEATSHEET?
